gui/CMakeLists.txt

```diff
@@ -9,13 +9,10 @@ CheckOptions:
   - { key: HeaderFileExtensions,          value: 'x' }
 ")
 
-    if (CMAKE_BUILD_TYPE STREQUAL "Release")
-        add_definitions(-DQT_NO_DEBUG)
-        add_definitions(-DQT_NO_DEBUG_OUTPUT)
-        add_definitions(-DQT_NO_WARNING_OUTPUT)
-    else()
-        add_definitions(-DQT_DEBUG)
-    endif()
+    add_compile_definitions($<$<NOT:$<CONFIG:Debug>>:-DQT_NO_DEBUG>)
```

Comment:
This does not account for `RelWithDebInfo` which did not set it before.

Reply:
Ah, I see you are generating a completely new solution which contains the GUI files as well. Nobody ever tested that and until we drop the provided solution file (which we most likely never will) I also consider this as not officially supported.

And adding this to the provided solution adds too much complexity which I am not willing to maintain - and I would also very much reject an external submission adding this as it would increase the build matrix complexity and if it needs to be adjusted it would break the build and the maintainer would end up somebody "within" project.